pkg/reconciler/v1alpha1/channel/channel.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package channel
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/golang/glog"
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName       = "channel-default-controller"
+	channelReconciled         = "ChannelReconciled"
+	channelUpdateStatusFailed = "ChannelUpdateStatusFailed"
+)
+
+type reconciler struct {
+	client        client.Client
+	restConfig    *rest.Config
+	dynamicClient dynamic.Interface
+	recorder      record.EventRecorder
+}
+
+// Verify the struct implements reconcile.Reconciler
+var _ reconcile.Reconciler = &reconciler{}
+
+// ProvideController returns a Channel controller.
+// This Channel controller is a default controller for channels of all provisioner kinds
+func ProvideController(mgr manager.Manager) (controller.Controller, error) {
+	// Setup a new controller to Reconcile channel
+	c, err := controller.New(controllerAgentName, mgr, controller.Options{
+		Reconciler: &reconciler{
+			recorder: mgr.GetRecorder(controllerAgentName),
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Watch channel events
+	// This controller is no-op when Channels are deleted
+	if err := c.Watch(
+		&source.Kind{Type: &v1alpha1.Channel{}},
+		&handler.EnqueueRequestForObject{},
+		predicate.Funcs{
+			DeleteFunc: func(event.DeleteEvent) bool {
+				return false
+			},
+		}); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+// Reconcile will check if the channel is being watched by provisioner's channel controller
+// This will improve UX. See https://github.com/knative/eventing/issues/779
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	glog.Infof("Reconciling channel %s", request)
```

Comment:
I didn't mean to do a blanket %v->%s replace, for example request is a struct, and I don't think it has a string method so will probably get logged not correctly.

Reply:
request has a String method that return NamespacedString which is what we need in all these messages- namespaced channel name

Reply:
I don't see String on Request, though I do see that NamespacedString has it.
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/reconcile/reconcile.go

But, because of inlining, it seems to use String from it :)